packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -47,13 +47,23 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform  {
           .childFile('contents.xcworkspacedata');
 
   /// The Xcode workspace (.xcworkspace directory) of the host app.
-  Directory get xcodeWorkspace => hostAppRoot.childDirectory('$_hostAppProjectName.xcworkspace');
+  Directory? get xcodeWorkspace {
+    final List<FileSystemEntity> contents = hostAppRoot.listSync();
```

Comment:
```
[module_test_ios] [STDOUT] stderr: Oops; flutter has exited unexpectedly: "FileSystemException: Directory listing failed, path = '/Volumes/Work/s/w/ir/x/t/flutter_module_test.5aHfpx/hello/macos/' (OS Error: No such file or directory, errno = 2)".
[module_test_ios] [STDOUT] stderr: A crash report has been written to /Volumes/Work/s/w/ir/x/t/flutter_module_test.5aHfpx/hello/flutter_01.log.
```
I think this needs to check if `hostAppRoot.exists`

Reply:
Added in https://github.com/flutter/flutter/pull/114099/commits/0305e86d45c696fdc21513c51703cb757a163bc4.